src/resources/assets/js/geo/components/imageMap.vue

```diff
@@ -106,17 +107,32 @@ export default {
         },
         // takes array of overlays as input and returns them as ol-tileLayers
         createOverlayTile(overlay) {
+            let top_left = fromLonLat([
+                overlay.attrs.top_left_lng,
+                overlay.attrs.top_left_lat,
+            ]);
+            let bottom_right = fromLonLat([
+                overlay.attrs.bottom_right_lng,
+                overlay.attrs.bottom_right_lat,
+            ]);
+
+            let projection = new Projection({
+                code: 'EPSG:3857',
+                units: 'degrees',
+                extent: [
+                    top_left[0],
+                    top_left[1],
+                    bottom_right[0],
+                    bottom_right[1],
+                ],
+            });
+
             return new TileLayer({
                 source: new ZoomifySource({
-                        url: this.overlayUrlTemplate.replaceAll(':id', overlay.id),
-                        size: [overlay.attrs.width, overlay.attrs.height],
-                        extent: [
-                            overlay.attrs.top_left_lng,
-                            overlay.attrs.bottom_right_lat,
-                            overlay.attrs.bottom_right_lng,
-                            overlay.attrs.top_left_lat,
-                        ],
-                        transition: 100,
+                    url: this.overlayUrlTemplate.replaceAll(':id', overlay.id),
+                    size: [overlay.attrs.width, overlay.attrs.height],
+                    transition: 100,
+                    projection: projection,
```

Comment:
I'm quite sure this is how the ZoomifySource is supposed to be created. The projection defines the extent, not the extent of the source (this basically only stores the size of the image). However, I could not get the projections right here. I loaded your GeoTIFF into QGIS and searched in BIIGLE at the location QGIS determined but the overlay was not displayed there. I also tried to make OpenLayers zoom to the extent of the tile layer but it showed some place in the Atlantic also without actually displaying the overlay. Maybe you have more luck figuring out the projections. Maybe there is still some error in the backend conversion?

This fixes the performance issues anyway. I think Openlayers tried to display the tile layer over the whole world because the extent definition was wrong.

Reply:
Great, thank you very much! It finally solves the performance issues.
I also verified the projection settings by loading the geoTIFF as well as the coordinate derived from `fromLonLat()` of the [createOverlayTile](https://github.com/biigle/geo/pull/47/files/2fc58509892e880b976903751b7227db609a9d49..ab7bc5e867aa00579e8e66e40e1453c0461969ef#diff-5973a87078606c3166889991a894c27918ec32c05de6ec3fd354c6df7aa47e19R110)-method into QGIS. The location of the geoTIFF and the projected coordinate match (see image below; red dot = coordinate derived from Biigle).
![pinpoint_location](https://github.com/user-attachments/assets/91e6b90e-e2b4-49db-8d42-a31e904b4025)

However, I can't see the tiled geoTIFF in Biigle either. I noticed in the network-tab that only **one** tile of each uploaded geoTIFF is actually loaded into the map. Also, the tiled .jpg-files of the geoTIFF's I tested look somewhat different to what the geoTIFF looks when uploaded in QGIS (in the test-cases they were black&white or entirely black rather than grayscale as in QGIS). I need to check why the jpeg's appear to be different and why they are not loaded/visualized properly in Biigle.

Reply:
> However, I can't see the tiled geoTIFF in Biigle either. I noticed in the network-tab that only one tile of each uploaded geoTIFF is actually loaded into the map.

You could check if the extend of the layer/source in OpenLayers matches the extent in QGIS.

> Also, the tiled .jpg-files of the geoTIFF's I tested look somewhat different to what the geoTIFF looks when uploaded in QGIS...

I saw that, too. The TIFF you cave me stored the data as floating point values. During tiling, these are converted to bytes, so lots of information is lost. We should add to the BIIGLE manual that users should convert their geoTIFFs to bytes before uploading them to BIIGLE.

Reply:
> You could check if the extent of the layer/source in OpenLayers matches the extent in QGIS.

@mzur, I used `.getTileGrid().getExtent()` on the ZoomifySource to see the extent and find out if it is set correctly. With your suggested settings (extent not specified), the extent is set automatically to [0, -height, width, 0] in pixel-values. However, since the TileLayer is projected to EPSG:3857, I wondered whether the extent of the source-layer had to be transformed as well. When I tried that (using `transformExtent`-method fom OL), OL tried to load tiles  that do not exist in the storage (e.g. _storage/geo-overlays/19/19_tiles/TileGroup255915/4-65400200-9503.jpg_) -- so it seems to be correct to let the extent be defined in pixel-values, as OL then tries to retrieve existing tiles at least.
However, the extent of the example geoTIFF in QGIS (and as a comparison, the extent from the VectorLayer which shows the geo-located images in the map) are both given in projected coordinates.

I can't really figure out what the issue is! Maybe you have another idea that could help me with that? Would be much appreciated!

Reply:
> Have you checked if the extent of the TIFF as reported in QGIS matches with what is imported into the BIIGLE DB?

Yes, I have checked the values. The geoTIFF gets (re)projected to _EPSG:4326_ when uploaded to BIIGLE. One example geoTIFF where I compared the extent as calculated in BIIGLE vs. QGIS:
1. `-2.9198048485707 | 57.0974838573358 | -2.9170062535908 | 57.0984589659731` BIIGLE
2. `-2.9213401128754679 | 57.0966521913032494 | -2.9184952039675358 | 57.0976265261187734` QGIS

The numbers are not identical but close (I suspect differences in the reprojection methods in QGIS vs. the [method](https://github.com/biigle/geo/pull/47/files#diff-93c2a8005ff37a47e89911d34cd4a939ca6e77d70077e4a8929f87fa03cef7eaR183) I wrote in BIIGLE).

Reply:
The difference might be quite significant if you convert it to m/km. One degree of latitude can be approximated with 111.1 km. The differences in the values you show can amount to 100-200 m offset on the map.

However, you should still see the tiled TIFF on the map. Maybe you can ask in the [OpenLayers discussions](https://github.com/openlayers/openlayers/discussions) if the implementation and projection configuration is correct.

Reply:
> However, you should still see the tiled TIFF on the map.

I somehow managed to visualize the geoTIFF on the map by trying different projections. It works using follwoing setup: `let projection = new Projection({ code: 'EPSG:4326', units: 'degrees' });` and setting the extent on the ZoomifySource to the EPSG:4326 encoded coordinates as they come from the DB. 
![Screenshot from 2024-10-01 11-27-08](https://github.com/user-attachments/assets/76f150f3-7604-493a-9651-39a3779c74ef)
It is not really intuitive to me, but I read the following sentence [here](https://openlayers.org/doc/tutorials/raster-reprojection.html) which might explain what happens: 

> If a source (based on ol/source/TileImage or ol/source/Image) has a projection different from the current ol/View’s projection then the reprojection happens automatically under the hood.

Reply:
Regarding this point:
> The TIFF you gave me stored the data as floating point values. During tiling, these are converted to bytes, so lots of information is lost. We should add to the BIIGLE manual that users should convert their geoTIFFs to bytes before uploading them to BIIGLE.

I spoke with Timm earlier today and showed him the current geoTIFF visualization (which quite obviously looks different to the QGIS visualization due to the information loss in the tiling process you mentioned). In addition, Tim said that such TIFF's contain lots of color-information which can be - and often need to be - adjusted by the user (e.g. have a look in the QGIS layer-properties tab --> Symbology). 
We discussed that the simplest solution for now would be to normalize the color band to the 0 - 255 range in the tiling process.

Reply:
> The difference might be quite significant if you convert it to m/km. One degree of latitude can be approximated with 111.1 km. The differences in the values you show can amount to 100-200 m offset on the map.

I checked the code of the geoTIFF-upload once again to see whether I had wrong calculations somewhere in there. Unfortunately, it seems like the projection error stems from the `transform()`-function of the Proj4php package. And I noticed that I'm not the only one having this [issue](https://github.com/proj4php/proj4php/issues/131).

Reply:
The projection: Doesn't the view also have EPSG:4326? Well, if it works, it works.

The precision: Ok, then we can normalize it. Vips should be able to handle that. We shuld also make this plain in the manual.

The transformation: Then we can't do anything about it. Maybe we could recommend a projection that doesn't need a transformation in the manual. This will always be exact. If the transformation done by BIIGLE produces too large an error, users will be able to do the transformation themselves before the upload.

Reply:
> The projection: Doesn't the view also have EPSG:4326? Well, if it works, it works.

The view is WGS84 / Pseudo-Mercator (EPSG:3857). 
Also, the solution proposed earlier did not really work (despite tiles showing up on the map). The projection was not not correctly implemented as explained in the answer to my [question](https://github.com/openlayers/openlayers/discussions/16253) I posed in the openlayers discussion panel. The key was to add coordinate transforms btw. the source and target-projection!